Source/RSCodeReaderViewController.swift

```diff
@@ -178,15 +178,17 @@ open class RSCodeReaderViewController: UIViewController, AVCaptureMetadataOutput
 			self.output.setMetadataObjectsDelegate(self, queue: queue)
 		}
 		// Remove previous added outputs from session
-		var metadataObjectTypes: [AnyObject]?
+		var metadataObjectTypes: [AVMetadataObject.ObjectType]?
 		for output in self.session.outputs {
-			metadataObjectTypes = (output as AnyObject).metadataObjectTypes as [AnyObject]?
-			self.session.removeOutput(output )
+			if let output = output as? AVCaptureMetadataOutput {
+				metadataObjectTypes = output.metadataObjectTypes
```

Comment:
I understand this was assigned previously as well, but wouldn't we want to append these objects to `metadataObjectTypes` instead? Otherwise, we can avoid the loop over the outputs (L182) and just use `self.session.outputs.last`.

Reply:
I was just cleaning the code, not the logic. Technically you're right, but it would be `self.session.outputs.last(where: { $0.metadataObjectTypes != nil })`. Since you're looping over all the outputs in order to remove them from the session I don't think you really lose anything by keeping it this way.

Reply:
> I was just cleaning the code, not the logic.

I understand, sorry for dragging you into this 😅

Tbh, I'm not very familiar with AVFoundation so I don't know if it's expected to have more than one output with `metadataObjectTypes != nil`. I just saw the diff when updating our dependencies and stumbled over this, purely from a code/language perspective, and thought I'd raise it. If you (or someone else) is confident that this doesn't actually cause potential issues, feel free to close this conversation.

Reply:
Me neither. I'd say that until somebody raises an issue with this code don't worry too much about it 😅